backend/pkg/factory/kafka/kafka.go

```diff
@@ -175,13 +177,22 @@ func NewKgoConfig(cfg config.Kafka, logger *zap.Logger, metricsNamespace string)
 						Extensions: kafkaSASLOAuthExtensionsToStrMap(cfg.SASL.OAUth.Extensions),
 					}, err
 				})
-			} else {
+			} else if cfg.SASL.OAUth.Token != "" {
 				mechanism = oauth.Auth{
 					Token:      cfg.SASL.OAUth.Token,
 					Extensions: kafkaSASLOAuthExtensionsToStrMap(cfg.SASL.OAUth.Extensions),
 				}.AsMechanism()
+			} else if cfg.SASL.OAUth.TokenFilePath != "" {
+				mechanism = oauth.Oauth(func(ctx context.Context) (oauth.Auth, error) {
+					token, err := os.ReadFile(cfg.SASL.OAUth.TokenFilePath)
```

Comment:
this should utilize filesystem service
https://github.com/redpanda-data/console/blob/master/backend/pkg/filesystem/service.go#L189

Reply:
GetFileByFilename

Reply:
I dont think we can use it because that is used to read the file that is cached. In this case, we cant really read from cache so that we get the most updated token from kubernetes. not the cached. @walbertus